File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& A) {

         int n = A.size();
    
    vector<vector<int>> B(n,vector<int>(n,0));
    
    for(int i = 0; i<n; i++){
        int col = n-1-i;
        for(int j=0;j<n;j++){
            
            B[j][col] = A[i][j];
        }
    }
   A= B;
    
        
    }
};

File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int starti=0,startj=0,endi=matrix.length-1,endj=endi;
        
        while(starti<endi){
            
            int i=starti,j=startj, n = endi-starti+1;
            while(i<endi){
                // System.out.println("first = "+matrix[starti][j]);
                //  System.out.println("second = "+ matrix[endi-(i-starti)][startj]);
                //  System.out.println("third = "+matrix[endi][endj-j]);
                //  System.out.println("fourth = "+matrix[i][endj]);
                int tmp = matrix[starti][j];
                matrix[starti][j] = matrix[endi-(i-starti)][startj];
                matrix[endi-i+starti][startj] = matrix[endi][endj-j+startj];
                matrix[endi][endj-j+startj] = matrix[i][endj];
                matrix[i][endj] = tmp;
                i++;
                j++;
            }
            
            starti++;
            startj++;
            endi--;
            endj--;
        }
        
    }
}